### PR TITLE
Fix bug: Trigger data protocol

### PR DESCRIPTION
**Pull request by danielmarchan3**
Add a new validation to the trigger data protocol

### CODE REVIEW

File: xmipp3/protocols/protocol_trigger_data.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def _defineParams(self, form):
                            '\n For this option, select the option send all items to output.')
         form.addParam('triggerProt', PointerParam,
                       pointerClass=self.getClassName(),
-                      condition='triggerSignal', allowsNull=True,
+                      condition='triggerSignal',
```
Comment (albertmena) [inline xmipp3/protocols/protocol_trigger_data.py:97]: Trigger data protocol does not appear in the GUI, is that correct?

File: xmipp3/protocols/protocol_trigger_data.py [hunk 2]
```diff
@@ -173,10 +173,9 @@ def _checkNewOutput(self):
             self.finished = False
 
         # Send the signal to the connected protocol
-        if self.triggerSignal.get():
-            if len(self.images) >= self.outputSize:
-                print('Sending signal to stop the input trigger data protocol')
-                self.stopWait()
+        if self.triggerSignal.get() and len(self.images) >= self.outputSize:
+            print('Sending signal to stop the input trigger data protocol')
```
Comment (albertmena) [inline xmipp3/protocols/protocol_trigger_data.py:177]: better self.info() than print()

File: xmipp3/protocols/protocol_trigger_data.py (hunk 1 above)
Comment (albertmena) [reply xmipp3/protocols/protocol_trigger_data.py:97]: Sorry, What I want to say is: 
" triggerProt" parameter does not appear in the 
Screenshot from 2022-10-28 11-50-54
window

Comment (danielmarchan3) [reply xmipp3/protocols/protocol_trigger_data.py:97]: That is correct, it only appears when the Send signal to a stop a stream is true.

**Review by albertmena — APPROVED** [review]

**Pull request merged**